src/block_cache/block_io.c

```diff
@@ -14,79 +14,158 @@
 WT_STAT_COMPR_RATIO_HIST_INCR_FUNC(ratio)
 
 /*
- * __wt_bt_read --
- *     Read a cookie referenced block into a buffer.
+ * __blkcache_read_corrupt --
+ *     Handle a failed read.
+ */
+static int
+__blkcache_read_corrupt(WT_SESSION_IMPL *session, int error, const uint8_t *addr, size_t addr_size,
+  const char *fail_msg) WT_GCC_FUNC_ATTRIBUTE((cold))
+{
+    WT_BM *bm;
+    WT_BTREE *btree;
+    WT_DECL_RET;
+
+    btree = S2BT(session);
+    bm = btree->bm;
+
+    ret = error;
+    WT_ASSERT(session, ret != 0);
+
+    F_SET(S2C(session), WT_CONN_DATA_CORRUPTION);
+    if (!F_ISSET(btree, WT_BTREE_VERIFY) && !F_ISSET(session, WT_SESSION_QUIET_CORRUPT_FILE)) {
+        WT_TRET(bm->corrupt(bm, session, addr, addr_size));
+        WT_RET_PANIC(session, ret, "%s: fatal read error: %s", btree->dhandle->name, fail_msg);
+    }
+    return (ret);
+}
+
+/*
+ * __wt_blkcache_read --
+ *     Read an address-cookie referenced block into a buffer.
  */
 int
-__wt_bt_read(WT_SESSION_IMPL *session, WT_ITEM *buf, const uint8_t *addr, size_t addr_size)
+__wt_blkcache_read(WT_SESSION_IMPL *session, WT_ITEM *buf, const uint8_t *addr, size_t addr_size)
 {
+    WT_BLKCACHE *blkcache;
+    WT_BLKCACHE_ITEM *blkcache_item;
     WT_BM *bm;
     WT_BTREE *btree;
+    WT_COMPRESSOR *compressor;
     WT_DECL_ITEM(etmp);
     WT_DECL_ITEM(tmp);
     WT_DECL_RET;
     WT_ENCRYPTOR *encryptor;
     WT_ITEM *ip;
     const WT_PAGE_HEADER *dsk;
     size_t compression_ratio, result_len;
-    const char *fail_msg;
+    uint64_t time_diff, time_start, time_stop;
+    bool blkcache_found, expect_conversion, found, skip_cache, timer;
 
+    blkcache = &S2C(session)->blkcache;
+    blkcache_item = NULL;
     btree = S2BT(session);
     bm = btree->bm;
-    fail_msg = NULL; /* -Wuninitialized */
+    compressor = btree->compressor;
+    encryptor = btree->kencryptor == NULL ? NULL : btree->kencryptor->encryptor;
+    blkcache_found = found = false;
+    skip_cache = blkcache->type == BLKCACHE_UNCONFIGURED;
 
```

Comment:
This is unintuitive. skip_cache is a boolean, but blkcache->type sounds like it is not, even though in our case it may only take the values of zero and one. In any case, it looks strange when they are part of the same assignment. Not incorrect, but rubs the wrong way.

Reply:
It's not an assignment of `blkcache.type`, it's an assignment of `skip_cache` based on the value of `blkcache.type`. If there's a way to clarify, or you want a comment, let me know.

Reply:
Ah, that's easy to miss. Especially after reading line 70, which primes you to expect something similar on line 71. 
This would be more clear if it read: skip_cache = (blkcache->type == BLKCACHE_UNCONFIGURED);

Up to you...

Reply:
Fixed as you suggested in 4e4041eb8.